### PR TITLE
[Merged by Bors] - chore: Remove `finset.sup_finset_image`

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/18893

---

- [x] depends on: leanprover-community/mathlib#18893
- [x] depends on: #3717

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Please update the SHAs for all files

bors d+

File: Mathlib/RingTheory/Finiteness.lean [hunk 1]
```diff
@@ -447,7 +447,7 @@ theorem fg_iff_compact (s : Submodule R M) : s.Fg ↔ CompleteLattice.IsCompactE
       -- Porting note: had to split this out of the `obtain`
       have := Finset.subset_image_iff.mp huspan
       obtain ⟨t, ⟨-, rfl⟩⟩ := this
-      rw [Finset.sup_finset_image, Function.comp.left_id, Finset.sup_eq_supᵢ, supr_rw, ←
+      rw [Finset.sup_image, Function.comp.left_id, Finset.sup_eq_supᵢ, supr_rw, ←
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Finiteness.lean:450]: You can't update the SHA in this file until I port leanprover-community/mathlib#18880

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3717
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3717~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib#18893~~
* ~~leanprover-community/mathlib4#3717~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**